app/src/main/res/layout/addfeed.xml

```diff
@@ -58,6 +58,17 @@
         </LinearLayout>
     </androidx.cardview.widget.CardView>
 
+    <Button
+        android:id="@+id/butRetry"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_centerHorizontal="true"
+        android:layout_margin="20dp"
+        android:text="@string/retry_label"
+        android:visibility="visible"
+        tools:background="@android:color/holo_red_light"
+        tools:visibility="visible" />
```

Comment:
You added the button but it looks like the button will always be displayed. This PR is missing some Java code that shows and hides the button. As the java code for updating the suggestions is located in `QuickFeedDiscoveryFragment`, it would probably be better to move the button to the fragment's layout, so it can be shown/hidden from there.

Reply:
Kindly do the rest, as this is my first PR, I am learning yet and would be a great source of motivation for me if this PR gets proper Java code by you and ultimately merged.

Reply:
Thank you for your kind-heartedness and patience.

Reply:
Hi @vishchi
We appreciate that you're willing to contribute to AntennaPod and eager to learn. However, asking our main developer to finish your work isn't really the way to go for a PR.
I'm sure @ByteHamster would be happy to point you in the right direction, though. Please let us know if you'd be interested in studying up on what's needed and completing the PR. :-) If not, that's perfectly fine as well of course - in that case we'll just close the PR (as there are other things that take priority at the moment).

Reply:
I am really sorry to ask out for that, but yeah, I want to learn to do the stuff. Please give me the direction. @ByteHamster

Reply:
I can't provide a full step-by-step guide because that would take more time than just doing the change myself (and I am pretty busy currently). The broad steps would be something like this:

- Find the layout of `QuickFeedDiscoveryFragment`
- Add the button to the layout (make sure that error message and button do not overlap)
- In `QuickFeedDiscoveryFragment`, create a button object and assign it from the layout file (just like the error message)
- Show and hide the button automatically if there is an error (just like the error message)
- Add an OnClickListener in `QuickFeedDiscoveryFragment` that retries loading (make sure to hide error message and retry button while loading, so that the progress indicator is visible)

If you have specific questions, feel free to ask. Please keep in mind that I can not teach you how to write Android apps in general. For that, there are tutorials available online. I can only help if it is something specific to AntennaPod.

Reply:
Thank you sir, working on it!

Reply:
Please close this PR, once after I do complete that thing, maybe I'll come up with a new PR and you can probably review that later.

Reply:
Okay, sure :) If you are not experienced with Android development, I suggest to start with a simple app that you write on your own. Understanding other people's source code (especially in big projects) is usually a bit harder than doing things your own way.